scanpy/tests/test_pca.py

```diff
@@ -72,27 +77,30 @@ def pca_params(array_type, svd_solver_type, zero_center):
     expected_warning = None
     svd_solver = None
     if svd_solver_type is not None:
-        if array_type is as_dense_dask_array:
+        # TODO: are these right for sparse?
+        if array_type in {as_dense_dask_array, as_sparse_dask_array}:
```

Comment:
Why is this changing?

Reply:
some parts of scanpy’s tests were using our own coercion, other parts anndata’s. We should be consistent unless we have a reason not to be.

Reply:
Oh, I actually meant "why change from list to set" but I did totally miss that more meaningful change.

I'm a little surprised that these things work. I'm also not sure these work for sparse.

Reply:
OK, so I’ll check if this just coerces to numpy array

Reply:
Dense dask arrays do appear to work. It looks like we will use `dask-ml` here.

Reply:
Sparse dask arrays don't seem to work. Shouldn't we be seeing an error from that?

Reply:
Perfect! This file uses `ARRAY_TYPES_SUPPORTED`, which marks sparse dask arrays as strict `xfail`. D.h. we’re testing that in-memory arrays and dense dask arrays work, and tests that sparse dask arrays throw an error.

Maybe this array should have a better name. Any suggestions?

Reply:
`as_sparse_dask_array`? I think it's fine. Or which array?

How do we start allowing sparse dask arrays if the parameter in `ARRAY_TYPES_SUPPORTED` marks them as xfail? Should we add the xfail here?

Reply:
Or which array?
> `ARRAY_TYPES_SUPPORTED`

>How do we start allowing sparse dask arrays
```diff
-ARRAY_TYPES_SUPPORTED
+ARRAY_TYPES
```

Once everything supports sparse dask arrays, we remove `ARRAY_TYPES_SUPPORTED` including its xfail marks

Reply:
For both things, fine for now, but okay to change later.